### PR TITLE
added some changes

### PR DESCRIPTION
**Pull request by asavienko**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -22,6 +26,14 @@
 .App-link {
   color: #61dafb;
 }
+.dynamic-delete-button {
+  cursor: pointer;
+  font-size: 17px;
+  transition: all .3s;
```
Comment (byuzyfyshyn) [inline src/App.css]: add mixin for transition

File: src/App.css [hunk 2]
```diff
@@ -22,6 +26,14 @@
...
+}
+.dynamic-delete-button:hover {
+  color: #666;
+}
 
 @keyframes App-logo-spin {
```
Comment (byuzyfyshyn) [inline src/App.css]: remove this part code

File: src/App.css (hunk 1 above)
Comment (byuzyfyshyn) [reply src/App.css]: or better move to styled component all this file

File: src/InputItem.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    this.state = {
+      todoInput: "",
+    }
+  }
+
+  onInputChange = (e) => {
+    this.setState({todoInput: e.target.value})
+  };
+  addInputToState = () => {
+    this.props.addTodoItem(this.state.todoInput);
```
Comment (byuzyfyshyn) [inline src/InputItem.js]:
```suggestion
    const {todoInput} = this.state;
    this.props.addTodoItem(todoInput);
```

File: src/InputItem.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+class InputItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoInput: "",
+    }
+  }
+
+  onInputChange = (e) => {
+    this.setState({todoInput: e.target.value})
```
Comment (byuzyfyshyn) [inline src/InputItem.js]:
```suggestion
    const todoInput = e.target.value;
    this.setState({todoInput});
```

File: src/Todo.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from "react"
+import InputItem from "./InputItem";
+import {ConfigProvider, Icon, List} from "antd";
+import {connect} from "react-redux";
+import TodoItem from "./TodoItem";
+
+
+const customizeRenderEmpty = () => (
+  <div style={{textAlign: 'center'}}>
```
Comment (byuzyfyshyn) [inline src/Todo.js]: remove inline style

File: src/Todo.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    <p style={{fontSize: 16}}>Nothing to do</p>
+  </div>
+);
+
+class Todo extends React.Component {
+
+  render() {
+    return (
+      <ConfigProvider renderEmpty={customizeRenderEmpty}>
+        {/*<ListControl/>*/}
```
Comment (byuzyfyshyn) [inline src/Todo.js]: remove comment

File: src/Todo.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        <List
+          header={<InputItem/>}
+          style={{
```
Comment (byuzyfyshyn) [inline src/Todo.js]: remove inline style

File: src/Todo.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            width: "550px",
+            background: "#fff",
+            display: "inline-block",
+            textAlign: "left",
+            marginTop: "25px"
+          }}
+          bordered
+          dataSource={this.props.todoList} // can I make filter without id in reducer?
+          renderItem={item => (<TodoItem
```
Comment (byuzyfyshyn) [inline src/Todo.js]: (item) || item in all places with only one parameter

File: src/TodoItem.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
+import React from "react"
+import {Icon, Input, List} from "antd";
+import styled from "styled-components";
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
import styled, {css} from "styled-components";
```

File: src/TodoItem.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+import {deleteTodoItem, editTodoItem, switchShowTodoItem} from "./actions/todoActions";
+import {connect} from "react-redux";
+import {StyledCheckbox} from "./AppStyle";
+
+const ListItemMeta = List.Item.Meta;
+
+const StyledListItemMeta = styled(ListItemMeta)`
+      text-decoration: ${({completedStyle}) => completedStyle ? "line-through" : "none"};
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: ${(props)=> props.completedStyle ? css`text-decoration: line-through; color: #999` : css`text-decoration: none `}

File: src/TodoItem.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+  deleteItem = (id) => {
+    this.props.deleteTodoItem(id)
+  };
+
+  switchShowCheckbox = (e, todoItemId) => {
+    this.props.switchShowTodoItem(todoItemId, e.target.checked)
+  };
+
+  toggleInput = (item) => {
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
  toggleInput = ({todoItem: editInputValue, todoItemId: editInputId}) => {
```

File: src/TodoItem.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    this.setState({
+      showEditInput: true,
+      editInputValue: item.todoItem,
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
      editInputValue,
```

File: src/TodoItem.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+      editInputId: item.todoItemId,
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
      editInputId,
```

File: src/TodoItem.js [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+    });
+  };
+
+  saveEditInputChanges = () => {
+    this.props.editTodoItem(
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: const {editInputValue, editInputId} = this.state;

File: src/TodoItem.js [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+            onChange={this.changeEditInput}
+            onBlur={this.saveEditInputChanges}
+            onPressEnter={this.saveEditInputChanges}
+          />
+        </List.Item>) ||
+      <List.Item
+        actions={[<Icon
+          className="dynamic-delete-button"
+          type="delete"
+          onClick={() => this.deleteItem(item.todoItemId)}
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: onClick={this.deleteItem}

File: src/TodoItem.js [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+    super(props);
+    this.state = {
+      showEditInput: false,
+      editInputValue: "",
+      editInputId: null,
+    }
+  }
+
+  deleteItem = (id) => {
+    this.props.deleteTodoItem(id)
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: const {todoItemId} = this.props.item

File: src/TodoItem.js [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      showEditInput: false,
+      editInputValue: "",
+      editInputId: null,
+    }
+  }
+
+  deleteItem = (id) => {
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
  deleteItem = () => {
```

File: src/TodoItem.js (hunk 16 above)
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
    this.props.deleteTodoItem(todoItemId)
```

File: src/TodoItem.js [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+      <List.Item
+        actions={[<Icon
+          className="dynamic-delete-button"
+          type="delete"
+          onClick={() => this.deleteItem(item.todoItemId)}
+        />]}
+      >
+        <StyledCheckbox
+          checked={item.todoItemActive}
+          onChange={(e) => this.switchShowCheckbox(e, item.todoItemId)}
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: same

File: src/TodoItem.js [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+          completedStyle={item.todoItemActive}
+          onClick={() => this.toggleInput(item)}
+        />
+      </List.Item>}
+    </React.Fragment>
+  }
+}
+
+
+const mapDispatchToProps = (dispatch) => {
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
const mapDispatchToProps = (dispatch, {item : {todoItemId, todoItemActive}}) => {
```

File: src/TodoItem.js [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+  return {
+    deleteTodoItem: (id) => {
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
    deleteTodoItem: () => {
```

File: src/TodoItem.js [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+      dispatch(deleteTodoItem(id))
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]:
```suggestion
      dispatch(deleteTodoItem(todoItemId))
```

File: src/actions/todoActions.js [hunk 22]
```diff
@@ -0,0 +1,54 @@
+import {updateTodoList} from "./todoPureActions";
+
+export const addTodoItem = (todoItem) => (dispatch, getState) => {
+  const state = getState();
+  const {todoReducer} = state;
```
Comment (byuzyfyshyn) [inline src/actions/todoActions.js]:
```suggestion
  const {todoReducer: {todoList}} = state;
```

File: src/actions/todoActions.js [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+  const newTodoList = todoReducer.todoList.concat([{
```
Comment (byuzyfyshyn) [inline src/actions/todoActions.js]: ### 
```suggestion
  const newTodoList = todoList.concat([{
```

**Review by byuzyfyshyn — CHANGES_REQUESTED** [review]

File: src/TodoItem.js (hunk 19 above)
Comment (asavienko) [reply src/TodoItem.js]: Function receives argument todoItemActive form state. Doesn't from porps.

File: src/InputItem.js [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+      this.setState({todoInput: ""});
+    }
+  };
+  switchCompletedAll = () => {
+    this.props.switchCompletedAll();
+    const allValueChecked = this.props.todoList.every(
+      ({todoItemActive}) => todoItemActive === true
+    );
+    this.setState({checkedAll: allValueChecked});
+    console.log(allValueChecked)
```
Comment (byuzyfyshyn) [inline src/InputItem.js]: remove console.log()

File: src/InputItem.js [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+    display: flex;
+    align-items: center;
+`;
+
+
+class InputItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoInput: "",
```
Comment (byuzyfyshyn) [inline src/InputItem.js:17]: rename todoInput to todoInputValue

File: src/InputItem.js [hunk 26]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {Input} from "antd";
+import {addTodoItem, switchCompletedAll} from "./actions/todoActions";
+import {connect} from "react-redux";
+import styled from "styled-components";
+import {StyledCheckbox} from "./AppStyle";
+
+const StyledWrapper = styled.div`
```
Comment (byuzyfyshyn) [inline src/InputItem.js:8]: rename StyledWrapper to StyledContainer

File: src/InputItem.js [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+    const {todoInput} = this.state;
+    if (todoInput.trim()) {
+      this.props.addTodoItem(todoInput);
+      this.setState({todoInput: ""});
+    }
+  };
+  switchCompletedAll = () => {
+    this.props.switchCompletedAll();
+    const allValueChecked = this.props.todoList.every(
+      ({todoItemActive}) => todoItemActive === true
```
Comment (byuzyfyshyn) [inline src/InputItem.js]: item => item.todoItemActive

File: src/InputItem.js [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+    );
+    this.setState({checkedAll: allValueChecked});
+    console.log(allValueChecked)
+  };
+
+  render() {
+    return <StyledWrapper>
+      <StyledCheckbox
+        checked={this.state.checkedAll}  /*The checkbox was worked not right, i made the controlled component,
+        but I need to do switchCompletedAll function when I click to child checkboxes in the list.
```
Comment (byuzyfyshyn) [inline src/InputItem.js:44]: remove comment

File: src/TodoItem.js [hunk 29]
```diff
@@ -0,0 +1,114 @@
...
+        />]}
+      >
+        <StyledCheckbox
+          checked={item.todoItemActive}
+          onChange={this.switchShowCheckbox}
+        />
+        <StyledListItemMeta
+          title={item.todoItem}
+          completedStyle={item.todoItemActive}
+          onClick={() => this.toggleInput(item)}
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: move item to function

File: src/TodoItem.js [hunk 30]
```diff
@@ -0,0 +1,114 @@
...
+
+const mapDispatchToProps = (dispatch, {item: {todoItemId}}) => {
+  return {
+    deleteTodoItem: () => {
+      dispatch(deleteTodoItem(todoItemId))
+    },
+    switchShowTodoItem: (todoItemActive) => {
+      dispatch(switchShowTodoItem(todoItemId, todoItemActive))
+    },
+    editTodoItem: (todoItem) => {
```
Comment (byuzyfyshyn) [inline src/TodoItem.js]: rename todoItem to todoItemValue

**Review by byuzyfyshyn — CHANGES_REQUESTED** [review]

**Pull request merged**